frontend/src/components/common/Header.tsx

```diff
@@ -1,57 +1,119 @@
-import { Button, Flex, Image } from "@chakra-ui/react";
-import React from "react";
+import { Button, Divider, Flex, Image } from "@chakra-ui/react";
+import React, { useContext, useState } from "react";
 import { useNavigate } from "react-router-dom";
 
-import Gear from "../../assets/greenGear.svg";
+import greenGear from "../../assets/greenGear.svg";
 import Logo from "../../assets/logo.png";
-import { HOME_PAGE, SETTINGS_PAGE } from "../../constants/Routes";
+import whiteGear from "../../assets/whiteGear.svg";
+import {
+  DASHBOARD_PAGE,
+  HOME_PAGE,
+  SETTINGS_PAGE,
+} from "../../constants/Routes";
+import AuthContext from "../../contexts/AuthContext";
+import Logout from "../auth/Logout";
 
 const Header = () => {
+  const { authenticatedUser } = useContext(AuthContext);
+
   const navigate = useNavigate();
 
+  const [isHovered, setIsHovered] = useState(false);
+
+  const handleHover = () => {
+    setIsHovered(true);
+  };
+
+  const handleMouseLeave = () => {
+    setIsHovered(false);
+  };
```

Comment:
nit: you can just move these inline, separate function isn't really needed unless there is a lot going on:
onMouseEnter={ () => setIsHovered(true)}

The advantage is that its easier to read what "onMouseEnter" does without having to click into a separate function.